src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
+const healthByDefault = 100;
+const decreaseOfHealth = 50;
+const deadBeast = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = healthByDefault) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isCarnivore = animal instanceof Carnivore;
+
+    if (!isCarnivore && !animal.hidden) {
+      animal.health -= decreaseOfHealth;
+    }
+
+    if (animal.health <= deadBeast) {
+      Animal.alive = Animal.alive.filter((beast) => beast.health > deadBeast);
+    }
```

Comment:
The `bite` method currently checks if the `animal` is not a `Carnivore` and not hidden before decreasing health. However, it should explicitly check if the `animal` is an instance of `Herbivore` to ensure that only herbivores are bitten. Consider changing the condition to `if (animal instanceof Herbivore && !animal.hidden)`.